src/qt/mempoolstats.cpp

```diff
@@ -0,0 +1,423 @@
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/mempoolstats.h>
+#include <qt/forms/ui_mempoolstats.h>
+
+#include <qt/clientmodel.h>
+#include <qt/guiutil.h>
+#include <stats/stats.h>
+
+static const char *LABEL_FONT = "Arial";
+static int LABEL_TITLE_SIZE = 22;
+static int LABEL_KV_SIZE = 12;
+
+static const int TEN_MINS = 600;
+static const int ONE_HOUR = 3600;
+static const int ONE_DAY = ONE_HOUR*24;
+
+static const int LABEL_LEFT_SIZE = 30;
+static const int LABEL_RIGHT_SIZE = 30;
+static const int GRAPH_PADDING_LEFT = 30+LABEL_LEFT_SIZE;
+static const int GRAPH_PADDING_RIGHT = 30+LABEL_RIGHT_SIZE;
+static const int GRAPH_PADDING_TOP = 10;
+static const int GRAPH_PADDING_TOP_LABEL = 150;
+static const int GRAPH_PADDING_BOTTOM = 50;
+static const int LABEL_HEIGHT = 15;
+
+void ClickableTextItem::mousePressEvent(QGraphicsSceneMouseEvent *event)
+{
+    Q_EMIT objectClicked(this);
+}
+
+void ClickableTextItem::setEnabled(bool state)
+{
+    if (state)
+        setDefaultTextColor(QColor(15,68,113, 250));
+    else
+        setDefaultTextColor(QColor(100,100,100, 200));
+}
+
+MempoolStats::MempoolStats(QWidget *parent) :
+QWidget(parent, Qt::Window),
+clientModel(0),
+titleItem(0),
+scene(0),
+drawTxCount(true),
+drawMinFee(false),
+drawDynMemUsage(true),
+timeFilter(TEN_MINS),
+ui(new Ui::MempoolStats)
+{
+    ui->setupUi(this);
+    if (parent) {
+        parent->installEventFilter(this);
+        raise();
+    }
+
+    // autoadjust font size
+    QGraphicsTextItem testText("jY"); //screendesign expected 27.5 pixel in width for this string
+    testText.setFont(QFont(LABEL_FONT, LABEL_TITLE_SIZE, QFont::Light));
+    LABEL_TITLE_SIZE *= 27.5/testText.boundingRect().width();
+    LABEL_KV_SIZE *= 27.5/testText.boundingRect().width();
+
+    scene = new QGraphicsScene();
+    ui->graphicsView->setScene(scene);
+    ui->graphicsView->setRenderHints(QPainter::Antialiasing | QPainter::SmoothPixmapTransform);
+
+    if (clientModel)
+        drawChart();
+}
+
+void MempoolStats::setClientModel(ClientModel *model)
+{
+    clientModel = model;
+
+    if (model)
+        connect(model, SIGNAL(mempoolStatsDidUpdate()), this, SLOT(drawChart()));
+}
+
+void MempoolStats::drawChart()
+{
+    if (!isVisible())
+        return;
+
+    if (!titleItem)
+    {
+        // create labels (only once)
+        titleItem = scene->addText(tr("Mempool Statistics"));
+        titleItem->setFont(QFont(LABEL_FONT, LABEL_TITLE_SIZE, QFont::Light));
+        titleLine = scene->addLine(0,0,100,100);
+        titleLine->setPen(QPen(QColor(100,100,100, 200), 2, Qt::SolidLine, Qt::RoundCap, Qt::RoundJoin));
+        
+        
+        QCheckBox *cb0 = new QCheckBox("Dynamic Memory Usage");
+        cb0->setStyleSheet("background-color: rgb(255,255,255);");
+        dynMemUsageSwitch = scene->addWidget(cb0);
+        connect(cb0, &QCheckBox::stateChanged, [cb0, this](){ drawDynMemUsage = cb0->isChecked(); drawChart(); });
```

Comment:
Move the user interacting bits to another function?

Also, do these really need to get redrawn every time? I'm not worried about performance, just code complexity.